src/api/app/controllers/webui/request_controller.rb

```diff
@@ -402,6 +384,25 @@ def accept_request
     end
   end
 
+  def prepare_bsrequest(action, params)
+    req = BsRequest.new(state: 'new', description: params[:description])
+    params.delete :description
+    klass_to_call = case action
+                    when :set_bugowner
+                      BsRequestActionSetBugowner
+                    when :delete
+                      BsRequestActionDelete
+                    when :add_role
+                      BsRequestActionAddRole
+                    when :change_devel
+                      BsRequestActionChangeDevel
+                    else
+                      raise APIException, 'method not found'
+                    end
+    req.bs_request_actions << klass_to_call.send(:new, params)
+    req
+  end
```

Comment:
This should be moved to a model.

Reply:
to which model @bgeuken ?

Reply:
BsRequestActions I would say. Though I just realized that you could do this instead:
req.bs_request_actions.create(type: action, $other_params)
```ruby

```